transaction-status/src/parse_token/extension/confidential_transfer.rs

```diff
@@ -0,0 +1,399 @@
+        }
+        ConfidentialTransferInstruction::WithdrawWithheldTokensFromMint => {
+            check_num_token_accounts(account_indexes, 4)?;
+            let withdraw_withheld_data: WithdrawWithheldTokensFromMintData =
+                *decode_instruction_data(instruction_data).map_err(|_| {
+                    ParseInstructionError::InstructionNotParsable(ParsableProgram::SplToken)
+                })?;
+            let proof_instruction_offset: i8 = withdraw_withheld_data.proof_instruction_offset;
+            let mut value = json!({
+                "mint": account_keys[account_indexes[0] as usize].to_string(),
+                "feeRecipient": account_keys[account_indexes[1] as usize].to_string(),
+                "instructionsSysvar": account_keys[account_indexes[2] as usize].to_string(),
+                "proofInstructionOffset": proof_instruction_offset,
+
+            });
+            let map = value.as_object_mut().unwrap();
+            parse_signers(
+                map,
+                3,
+                account_keys,
+                account_indexes,
+                "withdrawWithheldAuthority",
+                "multisigWithdrawWithheldAuthority",
+            );
+            Ok(ParsedInstructionEnum {
+                instruction_type: "withdrawWithheldConfidentialTransferTokensFromMint".to_string(),
+                info: value,
+            })
+        }
+        ConfidentialTransferInstruction::WithdrawWithheldTokensFromAccounts => {
+            let withdraw_withheld_data: WithdrawWithheldTokensFromAccountsData =
+                *decode_instruction_data(instruction_data).map_err(|_| {
+                    ParseInstructionError::InstructionNotParsable(ParsableProgram::SplToken)
+                })?;
+            let num_token_accounts = withdraw_withheld_data.num_token_accounts;
+            check_num_token_accounts(account_indexes, 4 + num_token_accounts as usize)?;
+            let proof_instruction_offset: i8 = withdraw_withheld_data.proof_instruction_offset;
+            let mut value = json!({
+                "mint": account_keys[account_indexes[0] as usize].to_string(),
+                "feeRecipient": account_keys[account_indexes[1] as usize].to_string(),
+                "instructionsSysvar": account_keys[account_indexes[2] as usize].to_string(),
+                "proofInstructionOffset": proof_instruction_offset,
+            });
+            let map = value.as_object_mut().unwrap();
+            let mut source_accounts: Vec<String> = vec![];
+            let first_source_account_index = account_indexes
+                .len()
+                .saturating_sub(num_token_accounts as usize);
+            for i in account_indexes[first_source_account_index..].iter() {
+                source_accounts.push(account_keys[*i as usize].to_string());
+            }
+            map.insert("sourceAccounts".to_string(), json!(source_accounts));
+            parse_signers(
+                map,
+                2,
```

Comment:
This should be `3`

Reply:
Awesome, thanks for spotting this!
At least it's wrong the right way (wouldn't cause a panic, just an incorrect multisig address)

Reply:
Hehe true :smile: